Hadrons/Modules/MDistil/Perambulator.hpp

```diff
@@ -157,93 +194,139 @@ void TPerambulator<FImpl>::setup(void)
 template <typename FImpl>
 void TPerambulator<FImpl>::execute(void)
 {
-    const DistilParameters &dp{ envGet(DistilParameters, par().DistilParams) };
     const int Nt{env().getDim(Tdir)};
+    auto &dilNoise = envGet(DistillationNoise<FImpl>, par().distilNoise);
+    int nNoise = dilNoise.size();	
+    int nVec=0;
+    if(par().nVec.empty())
+    {
+	nVec = dilNoise.getNl();
+    }
+    else
+    {
+        nVec = std::stoi(par().nVec);
+    }
+    int nDL = dilNoise.dilutionSize(DistillationNoise<FImpl>::Index::l);	
+    int nDS = dilNoise.dilutionSize(DistillationNoise<FImpl>::Index::s);	
+    int nDT = dilNoise.dilutionSize(DistillationNoise<FImpl>::Index::t);	
+    int nD = nDL * nDS * nDT;
+    auto &perambulator = envGet(PerambTensor, getName());
+    auto &epack = envGet(LapEvecs, par().lapEigenPack);
+    
+    pMode perambMode{par().perambMode};
+    LOG(Message)<< "Mode " << perambMode << std::endl;
 
     auto &solver=envGet(Solver, par().solver);
```

Comment:
This is the one thing I would have liked to do but could not find a solution for: Ideally, the code would look something like:

Solver solver; //this does not work
if(perambMode != pMode::inputSolve)
{
    solver = envGet(Solver, par().solver);
}

This works very similarly in line 267:
std::vector<FermionField> solveIn;
if(perambMode == pMode::inputSolve)
    {
        solveIn         = envGet(std::vector<FermionField>, par().unsmearedSolve);
    }

but there the constructor of std::vector allows an empty object that is changed at runtime, whereas Solver doesn't. I'm sure there is a solution, but I tried out a number of things and couldn't get anything to work.

The reason this would be desirable is that when running the inputSolve mode, the module still expects a solver module, even though it isn't used.